### PR TITLE
[RLlib] AlgorithmConfig cleanup 03: Cleaner names and structuring of API-stack config settings.

### PR DESCRIPTION
**Pull request by sven1977**
AlgorithmConfig cleanup 03:
* `AlgorithmConfig.experimental(_enable_new_api_stack=...)` renamed to `AlgorithmConfig.api_stack(enable_rl_module_and_learner)`.
* `env_runner_cls=SingleAgentEnvRunner|MultiAgentEnvRunner` automated to `AlgorithmConfig.api_stack(enable_env_runner_and_connector_v2=True)`
* Cleanup remaining already soft-deprecated `num_rollout_workers` and replace with the new `num_env_runners` setting.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]
I am just blindly trusting you Sven (as a doc reviewer)

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -663,10 +670,12 @@ def update_from_dict(
         # Namely, we want to re-instantiate the exploration config this config had
         # inside `self.experimental()` before potentially overwriting it in the
         # following.
-        if "_enable_new_api_stack" in config_dict:
-            self.experimental(
-                _enable_new_api_stack=config_dict["_enable_new_api_stack"]
-            )
+        enable_rl_module_and_learner = config_dict.get(
```
Comment (simonsays1980) [inline rllib/algorithms/algorithm_config.py:673]: Why is this? Imo users using still this "old" form must switch anyways in near future - so they could do it now. It's a small change.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Great change!

File: doc/source/rllib/doc_code/catalog_guide.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def __init__(self, *args, **kwargs):
 
 config = (
     PPOConfig()
-    .experimental(_enable_new_api_stack=True)
+    .api_stack(enable_rl_module_and_learner=True)
```
[the remark this answers is not in the document]
Comment (sven1977) [reply doc/source/rllib/doc_code/catalog_guide.py:116]: Yeah, me, too. `EnvRunners` growing to become adults :D

File: doc/source/rllib/doc_code/new_api_stack.py [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
-    # EnvRunner (replaces RolloutWorker) to use.
-    # Note that this step will be fully automated in the next release.
-    # Set the `env_runner_cls` to `SingleAgentEnvRunner` for single-agent setups and
-    # `MultiAgentEnvRunner` for multi-agent cases.
-    .env_runners(env_runner_cls=SingleAgentEnvRunner)
+    # Switch both the new API stack flags to True (both False by default).
+    # This enables the use of
+    # a) RLModule (replaces ModelV2) and Learner (replaces Policy)
+    # b) and automatically picks the correct EnvRunner (single-agent vs multi-agent)
+    # and enables ConnectorV2 support.
```
[the remark this answers is not in the document]
Comment (sven1977) [reply doc/source/rllib/doc_code/new_api_stack.py:12]: Yes, you can bring your own EnvRunner sub-class as you like.
See e.g. `rllib.algorithms.tests.test_worker_failures`, which uses sub-classes of `SingleAgentEnvRunner` and `MultiAgentEnvRunner`.

File: rllib/algorithms/algorithm_config.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/algorithm_config.py:669]: It's mostly for ourselves :| 
Some scripts of ours work for both API stacks and we don't want to make the `exploration_config` dependent on this. I'm fully with you: This is currently very hackish and should be cleaned up. I'll create another PR to see, what the impact would be for our code base (and tests), if we get rid of this hack.

**Pull request merged**